tfhe/src/integer/wopbs/mod.rs

```diff
@@ -22,6 +24,53 @@ pub struct WopbsKey {
     wopbs_key: crate::shortint::wopbs::WopbsKey,
 }
 
+#[must_use]
+pub struct IntegerWopbsLUT {
+    inner: WopbsLUTBase,
+}
+
+impl IntegerWopbsLUT {
+    pub fn new(small_lut_size: PlaintextCount, output_ciphertext_count: CiphertextCount) -> Self {
+        Self {
+            inner: WopbsLUTBase::new(small_lut_size, output_ciphertext_count),
+        }
+    }
+}
+
+impl TryFrom<Vec<Vec<u64>>> for IntegerWopbsLUT {
+    type Error = &'static str;
+
+    fn try_from(value: Vec<Vec<u64>>) -> Result<Self, Self::Error> {
+        let small_lut_size = value[0].len();
+        if !value.iter().all(|x| x.len() == small_lut_size) {
+            return Err("All small luts must have the same size");
+        }
+
+        let small_lut_count = value.len();
+
+        Ok(Self {
+            inner: WopbsLUTBase::from_vec(
+                value.into_iter().flatten().collect(),
+                CiphertextCount(small_lut_count),
+            ),
+        })
+    }
+}
+
+impl std::ops::Deref for IntegerWopbsLUT {
+    type Target = WopbsLUTBase;
+
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+impl std::ops::DerefMut for IntegerWopbsLUT {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.inner
+    }
+}
```

Comment:
Its probably fine for this but this kind of use of Deref is generally seen as an anti-pattern https://rust-unofficial.github.io/patterns/anti_patterns/deref.html

Reply:
ok got it, I prefer to avoid known anti patterns, which shape should it take, via a trait providing access to the underlying struct ?